### DISCUSSION TITLE
Data Fetching problem

### DISCUSSION THREAD

**Discussion by jaywei1234** [pocketbase/pocketbase#2129] [Q&A] [answered]
Hi, I'm working a project with Next12, I have a question about fetching data.

I can get the data isth getFullList
![image](https://user-images.githubusercontent.com/56056463/226958616-58d60fba-7316-4408-8be2-dc9b26bebd84.png)
 But after I change it to getList like this 
![image](https://user-images.githubusercontent.com/56056463/226958847-70c202f1-e4c7-414c-9cd3-4b7d2492d91b.png)
It shows data.map is not a function
![image](https://user-images.githubusercontent.com/56056463/226958959-fa53f633-e8eb-42c7-855e-7c58b8540fec.png)

**Comment by ganigeorgiev** [comment, marked as answer]
`getList` returns a paginated object with the following fields:

```js
{
    page:       number;
    perPage:    number;
    totalItems: number;
    totalPages: number;
    items:      Array<M>;
}
```

Or in other words, if you want to return only the records, you have to return the `records.items` prop (_I would suggest to rename the `records` variable to `result`, `data`, etc. to avoid eventual confusion_).

**Reply by jaywei1234** [reply]
Problem solved, thank you!!

**Reply by jaywei1234** [reply]
I found that I have the same problem of using getOne function in the dynamic page, and I don't know how to fix it.

**Reply by ganigeorgiev** [reply]
@jaywei1234 `getList()` and `getOne()` have different return result.

`getOne()` returns directly the record (aka. there is no items, totalPages, etc.).

If you are not sure what is the return value, you can always print it with `console.log()`.

**Reply by jaywei1234** [reply]
I think I know what the problem is. If I just do the data.map in dynamic page, it works. But it fails in passing the data to a component.
